### PR TITLE
Allow adding system properties to a Schedule

### PR DESCRIPTION
**Pull request by gokulavasan**
We need to pass Adapter Name as a system property to the program that is invoked through the scheduler.

This is required for couple of reasons:
i) To get the Adapter Name to set the tag in the Logging Context (which is currently achieved by piggybacking on schedule name)
ii) ScheduleTaskRunner needs to get the Adapter Name so that it can record the Runs under the Adapter Name (in addition to the Program name).

Bamboo : http://builds.cask.co/browse/CDAP-DUT1383

### CODE REVIEW

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/Scheduler.java [hunk 1]
```diff
@@ -45,6 +46,18 @@ public void schedule(Id.Program program, SchedulableProgramType programType, Sch
    *
    * @param program Program that needs to be run.
    * @param programType type of program.
+   * @param schedule Schedule with which the program runs.
+   * @param properties properties to be passed to the schedule
```
Comment (sagarkapare) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/Scheduler.java]: These are only system properties for the Schedule, which are different from the properties that are associated with the Schedule when its created ,right?

Comment (gokulavasan) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/Scheduler.java]: That is correct. Will fix the javadoc.

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/TimeScheduler.java [hunk 2]
```diff
@@ -119,14 +121,26 @@ void stop() throws SchedulerException {
...
   @Override
-  public void schedule(Id.Program programId, SchedulableProgramType programType, Schedule schedule)
+  public void schedule(Id.Program program, SchedulableProgramType programType, Schedule schedule)
     throws SchedulerException {
-    schedule(programId, programType, ImmutableList.of(schedule));
+    schedule(program, programType, schedule, ImmutableMap.<String, String>of());
+  }
+
+  @Override
+  public void schedule(Id.Program programId, SchedulableProgramType programType, Schedule schedule,
```
Comment (sagarkapare) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/TimeScheduler.java]: Maybe we can change the name of the parameter to `program` here as well?

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/TimeScheduler.java [hunk 3]
```diff
@@ -119,14 +121,26 @@ void stop() throws SchedulerException {
...
   }
 
   @Override
   public void schedule(Id.Program programId, SchedulableProgramType programType, Iterable<Schedule> schedules)
     throws SchedulerException {
+    schedule(programId, programType, schedules, ImmutableMap.<String, String>of());
+  }
+
+  @Override
+  public void schedule(Id.Program programId, SchedulableProgramType programType, Iterable<Schedule> schedules,
```
Comment (sagarkapare) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/TimeScheduler.java]: same here

**Comment by sagarkapare** [comment]
Minor comments, otherwise LGTM :+1:

**Comment by gokulavasan** [comment]
@sagarkapare Addressed comments. Will merge after build passes.

**Pull request merged**